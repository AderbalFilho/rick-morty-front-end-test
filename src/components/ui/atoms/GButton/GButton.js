import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './GButton.scss';

function GButton({ children, classes, ...rest }) {
  const className = classNames('g-button', classes);

  return (
    <button className={className} {...rest}>
      {children}
    </button>
  );
}

GButton.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
};

export default GButton;
