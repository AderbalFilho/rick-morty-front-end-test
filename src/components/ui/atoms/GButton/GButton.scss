@import 'styles/mixins';

.g-button {
  @include base-text;

  background-color: transparent;
  border: 2px solid var(--color-white);
  border-radius: var(--radius-medium);
  font-weight: bold;
  padding: 10px 16px 10px 17px;

  &:focus {
    outline: none;
  }
}
