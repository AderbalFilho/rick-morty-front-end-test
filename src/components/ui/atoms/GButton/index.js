import GButton from './GButton';

export default GButton;
