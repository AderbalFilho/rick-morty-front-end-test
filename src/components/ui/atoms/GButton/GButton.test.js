import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import GButton from './GButton';

describe('GButton', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<GButton>inGaia</GButton>);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  it('renders children', () => {
    const { container } = render(<GButton>inGaia</GButton>);
    const card = container.querySelector('.g-button');

    expect(card.textContent).toBe('inGaia');
  });

  it('renders classes', () => {
    const { container } = render(
      <GButton classes="button-search">inGaia</GButton>
    );
    const card = container.querySelector('.g-button');

    expect(card.classList).toContain('button-search');
  });
});
