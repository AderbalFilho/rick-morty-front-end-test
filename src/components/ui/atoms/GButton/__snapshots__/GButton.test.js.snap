// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GButton matches snapshot 1`] = `
<button
  className="g-button"
>
  inGaia
</button>
`;
